<a href="https://colab.research.google.com/github/Shreya-data/Play-Store-App-Reviews-Analysis/blob/main/play_store_app_reviews.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
import warnings
warnings.filterwarnings("ignore")
import ast

In [3]:
play_store_data=pd.read_csv("/content/drive/MyDrive/Play Store Data.csv")
user_review_data=pd.read_csv("/content/drive/MyDrive/User Reviews.csv")


---
**Explore Play Store Dataset**
---

In [4]:
play_store_data.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [5]:
play_store_data.columns.values

array(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'], dtype=object)

---
**COLUMN NAMES OF PLAY STORE DATA**
---
**App**- Name of the apps.

**Category**- Category under which the app falls.

**Rating**- Applications rating in playstore.

**Reviews**- Number of reviews given to apps.

**Size**- Size of the apps.

**Installs**- Number of installs of the apps.

**Type**- app is free or paid.

**Price**- Price of the apps.

**Content Rating** - Appropriate target audience of the apps.

**Genres**- Genres under which the app fall.

**Last** **updated**- Date when the app last updated.

**Current** **Ver**- Current version of the apps.

**Android** **Ver**- Minimum android version required for the 
apps to run.

In [6]:
print(f"There are total {play_store_data.shape[0]} rows and {play_store_data.shape[1]} columns in playstore dataframe.")

There are total 10841 rows and 13 columns in playstore dataframe.


In [7]:
play_store_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


In [8]:
play_store_data.describe(include="all")

App Category       Rating Reviews                Size    Installs  \
count    10841    10841  9367.000000   10841               10841       10841   
unique    9660       34          NaN    6002                 462          22   
top     ROBLOX   FAMILY          NaN       0  Varies with device  1,000,000+   
freq         9     1972          NaN     596                1695        1579   
mean       NaN      NaN     4.193338     NaN                 NaN         NaN   
std        NaN      NaN     0.537431     NaN                 NaN         NaN   
min        NaN      NaN     1.000000     NaN                 NaN         NaN   
25%        NaN      NaN     4.000000     NaN                 NaN         NaN   
50%        NaN      NaN     4.300000     NaN                 NaN         NaN   
75%        NaN      NaN     4.500000     NaN                 NaN         NaN   
max        NaN      NaN    19.000000     NaN                 NaN         NaN   

         Type  Price Content Rating Genres    Last Updated  \
count   10840  10841          10840  10841           10841   
unique      3     93              6    120            1378   
top      Free      0       Everyone  Tools  August 3, 2018   
freq    10039  10040           8714    842             326   
mean      NaN    NaN            NaN    NaN             NaN   
std       NaN    NaN            NaN    NaN             NaN   
min       NaN    NaN            NaN    NaN             NaN   
25%       NaN    NaN            NaN    NaN             NaN   
50%       NaN    NaN            NaN    NaN             NaN   
75%       NaN    NaN            NaN    NaN             NaN   
max       NaN    NaN            NaN    NaN             NaN   

               Current Ver Android Ver  
count                10833       10838  
unique                2832          33  
top     Varies with device  4.1 and up  
freq                  1459        2451  
mean                   NaN         NaN  
std                    NaN         NaN  
min                    NaN         NaN  
25%                    NaN         NaN  
50%                    NaN         NaN  
75%                    NaN         NaN  
max                    NaN         NaN

In [9]:
def PlaystoreInfo():
  PSinfo_df = pd.DataFrame(index=play_store_data.columns)
  PSinfo_df['Datatypes'] =  play_store_data.dtypes
  PSinfo_df['Count of non-null values'] = play_store_data.count()
  PSinfo_df['NaN values'] = play_store_data.isnull().sum()
  PSinfo_df['% NaN Values'] = (PSinfo_df['NaN values']/len(play_store_data)).round(4)*100 # or play_store_data.isnull().mean()
  PSinfo_df['Unique_count'] = play_store_data.nunique()
  return PSinfo_df
PlaystoreInfo() 


Datatypes  Count of non-null values  NaN values  % NaN Values  \
App               object                     10841           0          0.00   
Category          object                     10841           0          0.00   
Rating           float64                      9367        1474         13.60   
Reviews           object                     10841           0          0.00   
Size              object                     10841           0          0.00   
Installs          object                     10841           0          0.00   
Type              object                     10840           1          0.01   
Price             object                     10841           0          0.00   
Content Rating    object                     10840           1          0.01   
Genres            object                     10841           0          0.00   
Last Updated      object                     10841           0          0.00   
Current Ver       object                     10833           8          0.07   
Android Ver       object                     10838           3          0.03   

                Unique_count  
App                     9660  
Category                  34  
Rating                    40  
Reviews                 6002  
Size                     462  
Installs                  22  
Type                       3  
Price                     93  
Content Rating             6  
Genres                   120  
Last Updated            1378  
Current Ver             2832  
Android Ver               33

---
**Insights:**
---
From the above, we understand that except Rating columns, we are having a good dataset.

The number of null values are:

* **Rating** has 1474 null values which contributes 13.60% of the 
data.

* **Type** has 1 null value which contributes 0.01% of the data.

* **Content Rating** has 1 null value which contributes 0.01% of the data.

* **Current Ver** has 8 null values which contributes 0.07% of the data.

* **Android Ver** has 3 null values which contributes 0.03% of the data.

**We know that there is only one numeric type feature i.e. Rating.**

**Unique counts of Type and Content ratings are very small i.e. 3 and 6 respectively**

---
**Lets Explore The User Reviews Dataset**
---

In [10]:
user_review_data.head()

App                                  Translated_Review  \
0  10 Best Foods for You  I like eat delicious food. That's I'm cooking ...   
1  10 Best Foods for You    This help eating healthy exercise regular basis   
2  10 Best Foods for You                                                NaN   
3  10 Best Foods for You         Works great especially going grocery store   
4  10 Best Foods for You                                       Best idea us   

  Sentiment  Sentiment_Polarity  Sentiment_Subjectivity  
0  Positive                1.00                0.533333  
1  Positive                0.25                0.288462  
2       NaN                 NaN                     NaN  
3  Positive                0.40                0.875000  
4  Positive                1.00                0.300000

In [11]:
user_review_data.columns.values

array(['App', 'Translated_Review', 'Sentiment', 'Sentiment_Polarity',
       'Sentiment_Subjectivity'], dtype=object)

---
**COLUMN NAMES OF USER REVIEWS DATA**
---
**App** - The name of the application.

**Translated_Review** -Users' feedback is about the application.

**Sentiment** - tells us about a view or opinion of the user with respect to the application.

**Sentiment_Polarity** - Sentiment polarity for an element defines the orientation of the expressed sentiment, i.e., it determines if the text expresses the positive, negative or neutral sentiment of the user about the application.

**Sentiment_Subjectivity** - It refers to the text that contains text which is usually expressed by a human having typical moods, emotions, and feelings. Mostly it is a public opinion and not a factual information.


In [12]:
print(f"There are total {user_review_data.shape[0]} rows and {user_review_data.shape[1]} columns in User Reviews dataframe.")


There are total 64295 rows and 5 columns in User Reviews dataframe.


In [13]:
user_review_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 64295 entries, 0 to 64294
Data columns (total 5 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   App                     64295 non-null  object 
 1   Translated_Review       37427 non-null  object 
 2   Sentiment               37432 non-null  object 
 3   Sentiment_Polarity      37432 non-null  float64
 4   Sentiment_Subjectivity  37432 non-null  float64
dtypes: float64(2), object(3)
memory usage: 2.5+ MB


In [14]:
user_review_data.describe(include="all")

App Translated_Review Sentiment  Sentiment_Polarity  \
count                 64295             37427     37432        37432.000000   
unique                 1074             27994         3                 NaN   
top     Angry Birds Classic              Good  Positive                 NaN   
freq                    320               247     23998                 NaN   
mean                    NaN               NaN       NaN            0.182146   
std                     NaN               NaN       NaN            0.351301   
min                     NaN               NaN       NaN           -1.000000   
25%                     NaN               NaN       NaN            0.000000   
50%                     NaN               NaN       NaN            0.150000   
75%                     NaN               NaN       NaN            0.400000   
max                     NaN               NaN       NaN            1.000000   

        Sentiment_Subjectivity  
count             37432.000000  
unique                     NaN  
top                        NaN  
freq                       NaN  
mean                  0.492704  
std                   0.259949  
min                   0.000000  
25%                   0.357143  
50%                   0.514286  
75%                   0.650000  
max                   1.000000

In [15]:
def UserReviewInfo():
  URinfo_df = pd.DataFrame(index=user_review_data.columns)
  URinfo_df['Datatypes'] =  user_review_data.dtypes
  URinfo_df['Count of non-null values'] = user_review_data.count()
  URinfo_df['NaN values'] = user_review_data.isnull().sum()
  URinfo_df['% NaN Values'] = (URinfo_df['NaN values']/len(user_review_data)).round(4)*100 
  # or user_review_data.isnull().mean()
  URinfo_df['Unique_count'] = user_review_data.nunique()
  return URinfo_df
UserReviewInfo()

Datatypes  Count of non-null values  NaN values  \
App                       object                     64295           0   
Translated_Review         object                     37427       26868   
Sentiment                 object                     37432       26863   
Sentiment_Polarity       float64                     37432       26863   
Sentiment_Subjectivity   float64                     37432       26863   

                        % NaN Values  Unique_count  
App                             0.00          1074  
Translated_Review              41.79         27994  
Sentiment                      41.78             3  
Sentiment_Polarity             41.78          5410  
Sentiment_Subjectivity         41.78          4474

**Insights**
---

From the above, we understand that only for App column, we are having a good dataset.
There are so many of null values in dataset, we need to take care of that part.

The number of null values are:

* **Translated_Review** has 26868 null values which contributes 41.79% of the data.
* **Sentiment** has 26863 null values which contributes 41.78% of the data.
* **Sentiment_Polarity** has 26863 null values which contributes 41.78% of the data.
* **Sentiment_Subjectivity** has 26863 null values which contributes 41.78% of the data.

By Datatypes column, We know that there are two numeric type feature i.e. *Sentiment_Polarity* and *Sentiment_Subjectivity*.

Unique count of *Sentiment* are very small i.e. 3 .

**Data Cleaning - Univariate & Bivariate Analysis**
---

**Handling Missing Values in Play_Store_data**

In [16]:
play_store_data.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [17]:
PlaystoreInfo()

Datatypes  Count of non-null values  NaN values  % NaN Values  \
App               object                     10841           0          0.00   
Category          object                     10841           0          0.00   
Rating           float64                      9367        1474         13.60   
Reviews           object                     10841           0          0.00   
Size              object                     10841           0          0.00   
Installs          object                     10841           0          0.00   
Type              object                     10840           1          0.01   
Price             object                     10841           0          0.00   
Content Rating    object                     10840           1          0.01   
Genres            object                     10841           0          0.00   
Last Updated      object                     10841           0          0.00   
Current Ver       object                     10833           8          0.07   
Android Ver       object                     10838           3          0.03   

                Unique_count  
App                     9660  
Category                  34  
Rating                    40  
Reviews                 6002  
Size                     462  
Installs                  22  
Type                       3  
Price                     93  
Content Rating             6  
Genres                   120  
Last Updated            1378  
Current Ver             2832  
Android Ver               33

In [18]:
play_store_data.dropna(subset=["Rating","Type","Content Rating","Current Ver","Android Ver"],how="all").shape

(10841, 13)

from Playstoreinfo( ), we get to get know NAN values %

for **Rating** 13.60% of data are missing,

for **Type**	1% of data are missing,

for **Content Rating** 1% of data are missing,

for **Current Ver** 8% of data are missing,

for **Android Ver** 3% of data are missing.

Since 13.60% of data are missing in the Rating Column,

We cannot simply drop these NaN values of Rating Column.

**Handling Missing Values in user_reviews_data**

In [19]:
user_review_data.head()

App                                  Translated_Review  \
0  10 Best Foods for You  I like eat delicious food. That's I'm cooking ...   
1  10 Best Foods for You    This help eating healthy exercise regular basis   
2  10 Best Foods for You                                                NaN   
3  10 Best Foods for You         Works great especially going grocery store   
4  10 Best Foods for You                                       Best idea us   

  Sentiment  Sentiment_Polarity  Sentiment_Subjectivity  
0  Positive                1.00                0.533333  
1  Positive                0.25                0.288462  
2       NaN                 NaN                     NaN  
3  Positive                0.40                0.875000  
4  Positive                1.00                0.300000

In [20]:
UserReviewInfo()

Datatypes  Count of non-null values  NaN values  \
App                       object                     64295           0   
Translated_Review         object                     37427       26868   
Sentiment                 object                     37432       26863   
Sentiment_Polarity       float64                     37432       26863   
Sentiment_Subjectivity   float64                     37432       26863   

                        % NaN Values  Unique_count  
App                             0.00          1074  
Translated_Review              41.79         27994  
Sentiment                      41.78             3  
Sentiment_Polarity             41.78          5410  
Sentiment_Subjectivity         41.78          4474

In [21]:
new_user_df=user_review_data.dropna(subset=["Translated_Review","Sentiment","Sentiment_Polarity","Sentiment_Subjectivity"],how="all")

In [22]:
new_user_df.isnull().sum()

App                       0
Translated_Review         5
Sentiment                 0
Sentiment_Polarity        0
Sentiment_Subjectivity    0
dtype: int64

In [23]:
new_user_df.dropna(subset=["Translated_Review"],how="all",inplace=True)

In [24]:
print(new_user_df.isnull().sum())
print(new_user_df.shape)

App                       0
Translated_Review         0
Sentiment                 0
Sentiment_Polarity        0
Sentiment_Subjectivity    0
dtype: int64
(37427, 5)
